src/compiler/expressionToTypeNode.ts

```diff
@@ -711,7 +712,10 @@ export function createSyntacticTypeNodeBuilder(
                 undefined;
 
             if (type && !isConstTypeReference(type)) {
-                result = serializeExistingTypeNode(type, context);
+                const assertionNode = serializeExistingTypeNode(type, context);
+                if (resolver.canReuseTypeNodeAnnotation(context, node, getOriginalNode(assertionNode, isTypeNode), symbol)) {

```

Comment:
I was somewhat surprised that reusing an existing type node produces its copy with `.original` set on it. But also tbf I don't know when those copies are created in the first place so perhaps that's fine and expected. cc @dragomirtitian

Reply:
Copies are always created, this is expected. @weswigham might have better insight into why.